cloud/service_sg.go

```diff
@@ -110,9 +110,13 @@ func (s SecurityGroupService) RenameSecurityGroup(ctx context.Context, id int, r
 // CreateSecurityGroup creates a new security group.
 func (s SecurityGroupService) CreateSecurityGroup(ctx context.Context, req CreateSecurityGroupRequest) (*CreateSecurityGroupResponse, error) {
 	var resp CreateSecurityGroupResponse
-
 	p := s.path(SecurityGroupPath)
 
+	if req.Template == "" {
+		tmp := NewTemplate(req)
+		req.Template = tmp.String()
+	}
```

Comment:
I really like the idea of building a struct of a resource, e.g. security group, to then let the client create a template if one is not provided.